civitai-downloader-v2/src/api/services/download_service.py

```
@@ -0,0 +1,320 @@
+        
+        return {
+            "success": result.success,
+            "file_path": str(result.file_path) if result.file_path else None,
+            "error_message": result.error_message,
+            "task_id": result.task_id
+        }
+        
+    async def record_download_complete(self, task_id: str):
+        """Record completed download in database."""
+        task = self.download_manager.get_task_status(task_id)
+        if task and task.status == DownloadStatus.COMPLETED:
```

Ruff (0.12.2)

Undefined name `DownloadStatus`

(F821)